backend/eventloop/db_metrics_test.go

```diff
@@ -0,0 +1,161 @@
+
+			By("created four operation rows in database to test the count through the metrics")
+			OperationDB1 := db.Operation{
+				Operation_id:            "test-operation-1",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           "GitopsEngineInstance",
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: clusterAccess.Clusteraccess_user_id,
+				Last_state_update:       time.Now(),
+			}
+			err := dbq.CreateOperation(ctx, &OperationDB1, OperationDB1.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			OperationDB2 := db.Operation{
+				Operation_id:            "test-operation-2",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           "GitopsEngineInstance",
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: clusterAccess.Clusteraccess_user_id,
+				Last_state_update:       time.Now(),
+			}
+			err = dbq.CreateOperation(ctx, &OperationDB2, OperationDB2.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			err = dbq.GetOperationById(ctx, &OperationDB2)
+			Expect(err).To(BeNil())
+
+			OperationDB2.State = db.OperationState_In_Progress
+			err = dbq.UpdateOperation(ctx, &OperationDB2)
+			Expect(err).To(BeNil())
+
+			OperationDB3 := db.Operation{
+				Operation_id:            "test-operation-3",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           "GitopsEngineInstance",
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: clusterAccess.Clusteraccess_user_id,
+				Last_state_update:       time.Now(),
+			}
+			err = dbq.CreateOperation(ctx, &OperationDB3, OperationDB3.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			err = dbq.GetOperationById(ctx, &OperationDB3)
+			Expect(err).To(BeNil())
+
+			OperationDB3.State = db.OperationState_Completed
+			err = dbq.UpdateOperation(ctx, &OperationDB3)
+			Expect(err).To(BeNil())
+
+			OperationDB4 := db.Operation{
+				Operation_id:            "test-operation-4",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           "GitopsEngineInstance",
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: clusterAccess.Clusteraccess_user_id,
+				Last_state_update:       time.Now(),
+			}
+			err = dbq.CreateOperation(ctx, &OperationDB4, OperationDB4.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			err = dbq.GetOperationById(ctx, &OperationDB3)
+			Expect(err).To(BeNil())
+
+			OperationDB4.State = db.OperationState_Failed
+			err = dbq.UpdateOperation(ctx, &OperationDB4)
```

Comment:
Updating the operation here with the required states in the test because when I am trying to create operation with required state , It is always in Waiting state